inst/string.m

```diff
@@ -148,23 +149,30 @@
           error ('string: cell inputs must be cellstr');
         endif
         this.strs = in;
-        this.tfMissing = false (size (this.strs));
+        tfM = false (size (this.strs));
+        tfM(cellfun (@(x) isempty(x), this.strs)) = true;
+        this.tfMissing = tfM;
       elseif isnumeric (in)
         this.strs = arrayfun (@(x) {num2str(x)}, in);
-        this.tfMissing = false (size (this.strs));
+        tfM = false (size (this.strs));
+        tfM(arrayfun (@(x) isnan(x), in)) = true;
+        this.tfMissing = tfM;
       elseif isa (in, 'datetime')
         this.strs = arrayfun (@(x) {datestr(x)}, in);
         this.tfMissing = false (size (this.strs));
       elseif isa (in, 'duration') || isa (in, 'calendarDuration')
-        error ('string: duration and calendarDuration conversion are not implemented yet. Sorry.');
+        this.strs = arrayfun (@(x) dispstrs(x), in, "UniformOutput", false){:};
```

Comment:
Oh, thanks for implementing this one!

No need for `arrayfun` here either, I think: `strs = dispstrs(x)` is array-oriented, and returns an array of strings the same size as the input, with each `strs{i}` corresponding to `x(i)`.

For Tablicious's code style, I prefer `arrayfun (@(x) {blah(x)}, in)` to `arrayfun (@(x) blah(x), in, 'UniformOutput', false)`.

Reply:
> For Tablicious's code style, I prefer `arrayfun (@(x) {blah(x)}, in)` to `arrayfun (@(x) blah(x), in, 'UniformOutput', false)`.

These behave a bit differently, though